### PR TITLE
refactor: remove 'client.ToolchainObject'

### PR DESCRIPTION
**Pull request by xcoulon**
this PR goes one step further than https://github.com/codeready-toolchain/toolchain-common/pull/210: here, the `client.ToolchainObject` wrapper is totally removed since with the latest SDK upgrade, the `client.Object` (from `sigs.k8s.io/controller-runtime/pkg/client`) has all the methods that we need.

see also: 
- https://github.com/codeready-toolchain/member-operator/pull/298 
- https://github.com/codeready-toolchain/toolchain-e2e/pull/385
- https://github.com/codeready-toolchain/registration-service/pull/216

### CODE REVIEW

**Comment by xcoulon** [comment]
moving to WIP until the changes can be applied on `sandbox-sre` repo as well

File: pkg/client/object.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package client
+
+import (
+	"fmt"
+	"sort"
+
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type CompareToolchainObjects func(firstObject, secondObject runtimeclient.Object) (bool, error)
```
Comment (MatousJobanek) [inline pkg/client/object.go:10]: is this type still used anywhere? It doesn't seem so.

**Review by MatousJobanek — COMMENTED** [review]
Nice cleanup :+1: 
Shouldn't be also renamed all the related functions/types so they don't contain the removed `ToolchainObject` type name? The same applies also to the comments.

File: pkg/client/object.go (hunk 1 above)
Comment (xcoulon) [reply pkg/client/object.go:10]: > Nice cleanup 👍
> Shouldn't be also renamed all the related functions/types so they don't contain the removed `ToolchainObject` type name? The same applies also to the comments.

Actually, I did such renaming in sandbox-sre (PR coming soon), but I may have forgotten to do it in a few other places. Let me check that!

Comment (xcoulon) [reply pkg/client/object.go:10]: (looks like my response to your comment landed at the wrong location 😮 )

Comment (xcoulon) [reply pkg/client/object.go]: done in 573a24b

File: pkg/client/object.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
+package client
+
+import (
+	"fmt"
+	"sort"
+
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type CompareObjects func(firstObject, secondObject runtimeclient.Object) (bool, error)
```
Comment (MatousJobanek) [inline pkg/client/object.go:10]: So, is this really used somewhere? if so, then where? 
https://github.com/codeready-toolchain/toolchain-common/pull/211#discussion_r716450286

File: pkg/client/object.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	toolchainObjects := make([]runtimeclient.Object, len(objects))
+	for i, name := range names {
+		for _, object := range objects {
+			if fmt.Sprintf("%s,%s", object.GetNamespace(), object.GetName()) == name {
+				toolchainObjects[i] = object
+				break
+			}
+		}
+	}
+	return toolchainObjects
```
Comment (MatousJobanek) [inline pkg/client/object.go:31]:
```suggestion
	sortedObjects := make([]runtimeclient.Object, len(objects))
	for i, name := range names {
		for _, object := range objects {
			if fmt.Sprintf("%s,%s", object.GetNamespace(), object.GetName()) == name {
				sortedObjects[i] = object
				break
			}
		}
	}
	return sortedObjects
```

**Review by MatousJobanek — APPROVED** [review]
Thanks for renaming it :+1:

File: pkg/client/object.go (hunk 2 above)
Comment (xcoulon) [reply pkg/client/object.go:10]: indeed, it's not (I forgot to check and delete it)

Comment (xcoulon) [reply pkg/client/object.go]: removed in 0676b94

**Pull request merged**